mkosi/config.py

```diff
@@ -1975,6 +1977,49 @@ class Config:
 
     image: Optional[str]
 
+    def finalize_environment(self) -> dict[str, str]:
+        env = {
+            "SYSTEMD_TMPFILES_FORCE_SUBVOL": "0",
+            "SYSTEMD_ASK_PASSWORD_KEYRING_TIMEOUT_SEC": "infinity",
+            "SYSTEMD_ASK_PASSWORD_KEYRING_TYPE": "session",
+            "TERM": finalize_term(),
+        }
+
+        if self.image is not None:
+            env["SUBIMAGE"] = self.image
+        if self.image_id is not None:
+            env["IMAGE_ID"] = self.image_id
+        if self.image_version is not None:
+            env["IMAGE_VERSION"] = self.image_version
+        if self.source_date_epoch is not None:
+            env["SOURCE_DATE_EPOCH"] = str(self.source_date_epoch)
+        if self.proxy_url is not None:
+            for e in ("http_proxy", "https_proxy"):
+                env[e] = self.proxy_url
+                env[e.upper()] = self.proxy_url
+        if self.proxy_exclude:
+            env["no_proxy"] = ",".join(self.proxy_exclude)
+            env["NO_PROXY"] = ",".join(self.proxy_exclude)
+        if self.proxy_peer_certificate:
+            env["GIT_PROXY_SSL_CAINFO"] = "/proxy.cacert"
+        if self.proxy_client_certificate:
+            env["GIT_PROXY_SSL_CERT"] = "/proxy.clientcert"
+        if self.proxy_client_key:
+            env["GIT_PROXY_SSL_KEY"] = "/proxy.clientkey"
+        if dnf := os.getenv("MKOSI_DNF"):
+            env["MKOSI_DNF"] = dnf
+        if gnupghome := os.getenv("GNUPGHOME"):
+            env["GNUPGHOME"] = gnupghome
```

Comment:
If think this change i a good cleanup, but we should document the undocumented variables:
- the four variables from `env` should just be noted, that mkosi sets them for internal use, the same should maybe be done for `MKOSI_DNF` or maybe all `MKOSI_` variables? 
- `SUBIMAGE` is not yet documented (and used?)
- all proxy variables are not documented
- that we forward `GNUPGHOME` from the environment is also undocumented

Reply:
Most of these are just for internal use and users don't need to know about them I think, let's leave this for another day.